WiinUPro/Controls/NintyControl.xaml.cs

```diff
@@ -394,6 +394,32 @@ public void LoadProfile(string fileName)
 
             if (loadedProfile != null)
             {
+                if (fileName.Length > 6 & fileName[fileName.Length - 6] == '_')
+                {
+                    int n = (int)Char.GetNumericValue(fileName[fileName.Length - 5]);
+                    bool changeProfile = false;
+                    if (n == 1 && ScpDirector.Access.IsConnected(ScpDirector.XInput_Device.Device_A))
+                    {
+                        changeProfile = true;
+                    }
+                    else if (n == 2 && ScpDirector.Access.IsConnected(ScpDirector.XInput_Device.Device_B))
+                    {
+                        changeProfile = true;
+                    }
+                    else if (n == 3 && ScpDirector.Access.IsConnected(ScpDirector.XInput_Device.Device_C))
+                    {
+                        changeProfile = true;
+                    }
+                    else if (n == 4 && ScpDirector.Access.IsConnected(ScpDirector.XInput_Device.Device_D))
+                    {
+                        changeProfile = true;
+                    }

```

Comment:
A way to iterate through the connected devices would simplify this code. Such iteration would also simplify the code relating to the connecting of these devices in MainWindow.xaml.cs.